agent/agent.go

```diff
@@ -729,6 +743,10 @@ func (a *Agent) Start(ctx context.Context) error {
 			}
 		}
 	} else {
+		if a.baseDeps.UseV2Resources() {
+			return fmt.Errorf("can't start agent: client agents are not supported with v2 resources")
+		}
+
```

Comment:
This is irrelevant to this PR, but I thought it would be good to throw an error if you try to use the V2 catalog in client mode for now.